packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx

```diff
@@ -120,144 +177,73 @@ const ClinicalResultBloodPressure = (props) => {
-  } else {
-    const compareUnits = {};
-    const compareStatusIsInError = {};
-
-    const compareConceptDisplays = {
-      originalSystolic: systolic && systolic.conceptDisplay,
-      originalDiastolic: diastolic && diastolic.conceptDisplay,
-    };
-    const compareDatetimeDisplays = {
-      originalSystolic: systolic && systolic.datetimeDisplay,
-      originalDiastolic: diastolic && diastolic.datetimeDisplay,
-    };
+    return <ResultError />;
+  }
+  if (hasResultNoData) {
+    return <NoData />;
+  }
 
-    let iconGroupDisplayElement = null;
+  const systolicResult = createResult(systolic);
+  const diastolicResult = createResult(diastolic);
 
-    const noDataSystolic = (systolic && systolic.resultNoData === true);
-    if (systolic && !noDataSystolic) {
-      const {
-        result,
-        status,
-        conceptDisplay,
-        datetimeDisplay,
-      } = systolic;
+  const conceptDisplayElement = createConceptDisplays({
+    originalSystolic: systolicResult.conceptDisplay,
+    originalDiastolic: diastolicResult.conceptDisplay,
+    systolic: systolicResult.cleanedConceptDisplay,
+    diastolic: diastolicResult.cleanedConceptDisplay,
+  });
 
-      if (!isEmpty(result.unit)) { compareUnits.systolic = result.unit.trim().toLowerCase(); }
-      if (!isEmpty(status)) { compareStatusIsInError.systolic = checkIsStatusInError(status); }
-      if (!isEmpty(conceptDisplay)) { compareConceptDisplays.systolic = conceptDisplay.trim().toLowerCase(); }
-      if (!isEmpty(datetimeDisplay)) { compareDatetimeDisplays.systolic = datetimeDisplay.trim().toLowerCase(); }
-    }
+  const datetimeDisplayElement = createDatetimeDisplays({
+    originalSystolic: systolicResult.datetimeDisplay,
+    originalDiastolic: diastolicResult.datetimeDisplay,
+    systolic: systolicResult.cleanedDatetimeDisplay,
+    diastolic: diastolicResult.cleanedDatetimeDisplay,
+  });
 
-    const noDataDiastolic = (diastolic && diastolic.resultNoData === true);
-    if (diastolic && !noDataDiastolic) {
-      const {
-        result,
-        status,
-        conceptDisplay,
-        datetimeDisplay,
-      } = diastolic;
+  const hasModifiedIcon = (systolicResult.isModified) || (diastolicResult.isModified);
+  const hasCommentIcon = (systolicResult.hasComment) || (diastolicResult.hasComment);
+  const hasUnverifiedIcon = (systolicResult.isUnverified) || (diastolicResult.isUnverified);
 
-      if (!isEmpty(result.unit)) { compareUnits.diastolic = result.unit.trim().toLowerCase(); }
-      if (!isEmpty(status)) { compareStatusIsInError.diastolic = checkIsStatusInError(status); }
-      if (!isEmpty(conceptDisplay)) { compareConceptDisplays.diastolic = conceptDisplay.trim().toLowerCase(); }
-      if (!isEmpty(datetimeDisplay)) { compareDatetimeDisplays.diastolic = datetimeDisplay.trim().toLowerCase(); }
-    }
+  let iconGroupDisplayElement = null;
+  const decoratedResultDisplay = [];
+  if (systolic || diastolic) {
```

Comment:
should we have checked that  at least systolic or diastolic was provided above and if not, should we have` returned <NoData />;` ?

Reply:
Also, should we update this to use `systolicResult` and `diastolicResult` since these are the sanitized result objects we are using to render content with?

Reply:
any update? or did I miss the changes?

Reply:
Sorry addressed here, https://github.com/cerner/terra-clinical/pull/648/commits/415228472aa9b535d58726d4a69c09986fa23e4d#diff-13926e711e02256769a4612c0d911268R125
Discussed with Neil, we should just return a singular `--` if there is neither provided and not nothing like we currently do. Also added it to the no data test.